sympy/core/power.py

```diff
@@ -22,8 +22,10 @@
 
 def isqrt(n):
     """Return the largest integer less than or equal to sqrt(n)."""
-    if n < 17984395633462800708566937239552:
-        return int(_sqrt(n))
+    if n < 4503599761588224:
+        s = int(_sqrt(n))
```

Comment:
There are a couple of minor corner cases to consider here:

- If the incoming `n` is not an integer, then `int(_sqrt(n))` won't necessarily give the right result even for small `n`. Using `int(_sqrt(int(n)))` would cover that case. On the other hand, the check will catch any bad results anyway, so it doesn't matter much. On the third hand, the check itself could be somewhat expensive for non-integer `n`, and we end up converting `n` to an integer using `int(n)` for the slow branch anyway, so doing an initial `int(n)` in this branch is probably a good idea.

- If `n` is very negative (`n <= 2**970 - 2**1024`), then this branch will raise `OverflowError` as a result of the implicit int-to-float conversion overflowing; for compatibility with the slow version, we might want `ValueError` instead.